core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -227,10 +221,10 @@ public void delete(Delete delete) throws CrudException {
         read(key, createGet(key));
       }
       mutationConditionsValidator.checkIfConditionIsSatisfied(
-          delete, snapshot.getFromReadSet(key).orElse(null));
+          delete, snapshot.getResult(key).orElse(null));
```

Comment:
Ditto. Use the added `getResult()` method to use the result that's merged with a mutation for the same record in the write set for conditional deletes.